src/components/AddAssignment.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { Link } from 'react-router-dom'
+import Cookies from 'js-cookie';
+import Button from '@mui/material/Button';
+import Radio from '@mui/material/Radio';
+import {DataGrid} from '@mui/x-data-grid';
+import {SERVER_URL} from '../constants.js';
+import TextField from '@mui/material/TextField';
+
+// NOTE:  for OAuth security, http request must have
+//   credentials: 'include' 
+//
+
+class AddAssignment extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {courseId: '', assignmentName: '', dueDate: ''};
+      this.handleChange = this.handleChange.bind(this);
+      this.handleSubmit = this.handleSubmit.bind(this);
+    };
+ 
+    handleChange(event) {
+      const target = event.target;
+      const value = target.value;
+      const name = target.name;
+      this.setState({[name]: value});
+    }
+
+    handleSubmit(event) {
+      event.preventDefault();
+      const token = Cookies.get('XSRF-TOKEN');
+      fetch(`${SERVER_URL}gradebook`,
```

Comment:
export const SERVER_URL = 'http://localhost:8081'
```suggestion
      fetch(`${SERVER_URL}/gradebook`,
```